pkg/sql/catalog/rewrite/rewrite.go

```diff
@@ -651,6 +651,16 @@ func rewriteSchemaChangerState(
 	var droppedConstraints catalog.ConstraintIDSet
 	for i := 0; i < len(state.Targets); i++ {
 		t := &state.Targets[i]
+		// TableData elements are special and will be re-written directly.
+		if data := t.GetTableData(); data != nil {
+			rewrite, ok := descriptorRewrites[data.TableID]
+			if !ok {
+				return errors.Errorf("missing rewrite for id %d in %s", data.TableID, screl.ElementString(t.Element()))
+			}
+			data.TableID = rewrite.ID
+			data.DatabaseID = rewrite.ParentID
+			continue
+		}
```

Comment:
This rewrite makes sense but should this be in the switch case below? Not a rethorical question. Or is this case somehow different? If so I can't see it and this deserves commentary perhaps.

Reply:
It's because the parentID is not contained within the map that we are using, so it ends up special cased. Let me add the comment:
		// Since the parent database ID is never written in the descriptorRewrites
		// map we need to special case certain elements that need their ParentID
		// re-written.